apps/backend/src/libs/modules/ec2-cloudformation/ec2-cloudformation.module.ts

```diff
@@ -0,0 +1,129 @@
+import {
+	CloudFormationClient,
+	CreateStackCommand,
+	DeleteStackCommand,
+	DescribeStacksCommand,
+	waitUntilStackCreateComplete,
+	waitUntilStackDeleteComplete,
+} from "@aws-sdk/client-cloudformation";
+
+import { type Logger } from "../logger/logger.js";
+import {
+	Capability,
+	OutputKey,
+	ParameterKey,
+	StackPrefix,
+} from "./libs/enums/enums.js";
+
+type Constructor = {
+	accessKeyId: string;
+	logger: Logger;
+	region: string;
+	secretAccessKey: string;
+	templateBody: string;
+};
+
+type CreateEC2Parameters = {
+	imageId: string;
+	meetingId: string;
+	securityGroupId: string;
+};
+
+class CloudFormationEC2 {
+	#client: CloudFormationClient;
+	#logger: Logger;
+	#templateBody: string;
+
+	constructor({
+		accessKeyId,
+		logger,
+		region,
+		secretAccessKey,
+		templateBody,
+	}: Constructor) {
+		this.#templateBody = templateBody;
+		this.#logger = logger;
+
+		this.#client = new CloudFormationClient({
+			credentials: {
+				accessKeyId: accessKeyId,
+				secretAccessKey: secretAccessKey,
+			},
+			region,
+		});
+	}
+
+	async #getInstanceIdFromStack(stackName: string): Promise<string> {
```

Comment:
```suggestion
	private async getInstanceIdFromStack(stackName: string): Promise<string> {
```

Use the same approach for all

Reply:
hey, ts run on dev and compile to the js so private functions and variables convert to public.  I change # to private same to variables.

Reply:
I think we don't need to worry about the case you mentioned. Even if ts compiles private methods to public, it will not allow us to compile if we try to use private methods outside the class